pkg/bpf/map_linux.go

```diff
@@ -226,17 +229,25 @@ func (m *Map) WithCache() *Map {
 	return m
 }
 
-func (m *Map) GetFd() int {
-	return m.fd
+// GetFd returns the file descriptor of the BPF map
+func (m *Map) GetFd() (fd int) {
+	m.lock.RLock()
+	fd = m.fd
+	m.lock.RUnlock()
+	return
```

Comment:
ICYMI, the approach below is now optimized by the Go compiler to "do the right thing" (efficiently):

```
m.RLock()
defer m.RUnlock()
return m.fd
```

(I don't care that much if it's used here or not so feel free to disregard this comment)